pkgs/development/python-modules/ruffus/default.nix

```diff
@@ -9,45 +9,39 @@
 
 buildPythonPackage rec {
   pname = "ruffus";
-  version = "2.8.1";
+  version = "2.8.4";
 
   src = fetchFromGitHub {
     owner = "cgat-developers";
     repo = pname;
     rev = "v${version}";
-    sha256 = "1gyabqafq4s2sy0prh3k1m8859shzjmfxr7fimx10liflvki96a9";
+    sha256 = "0fnzpchwwqsy5h18fs0n90s51w25n0dx0l74j0ka6lvhjl5sxn4c";
   };
 
   propagatedBuildInputs = [ gevent ];
 
-  postPatch = ''
-    sed -i -e 's|/bin/bash|${stdenv.shell}|'          ruffus/test/Makefile
-    sed -i -e 's|\tpytest|\t${pytest}/bin/pytest|'    ruffus/test/Makefile
-    sed -i -e 's|\tpython|\t${python.interpreter}|'   ruffus/test/Makefile
-    sed -i -e 's|/usr/bin/env bash|${stdenv.shell}|'  ruffus/test/run_all_unit_tests.cmd
-    sed -i -e 's|python3|${python.interpreter}|'      ruffus/test/run_all_unit_tests3.cmd
-    sed -i -e 's|python %s|${python.interpreter} %s|' ruffus/test/test_drmaa_wrapper_run_job_locally.py
-  '';
-
-  makefile = "ruffus/test/Makefile";
-
   checkInputs = [
-    gevent
     hostname
     pytest
   ];
 
+  # test files do indeed need to be executed separately
   checkPhase = ''
-    export HOME=$TMPDIR
-    cd ruffus/test
-    make all PYTEST_OPTIONS="-q --disable-warnings"
+    pushd ruffus/test
+    rm test_with_logger.py  # spawns 500 processes
+    for f in test_*.py ; do
+      HOME=$TMPDIR pytest -v --disable-warnings $f
+    done
+    popd
   '';
 
   meta = with stdenv.lib; {
     description = "Light-weight Python Computational Pipeline Management";
     homepage = "http://www.ruffus.org.uk";
     license = licenses.mit;
     maintainers = [ maintainers.kiwi ];
+    # tests very flaky & hang often
+    broken = stdenv.isDarwin;
```

Comment:
does the package itself hang during normal usage? or only the test suite?

Reply:
Just the test suite, but I don't have a realistic example to exercise this package properly.

Reply:
I would say enable it, and just disable tests on darwin. If a darwin user does have an issue, hopefully they will create an issue with steps to reproduce error.

Reply:
Ok. Suppose it's not so bad because we're not really hacking this package to get it installed and the user is getting something very similar to what they would get if they installed using `pip`.